spec/generate_windows_spec.sh

```diff
@@ -65,7 +65,7 @@ echo
 for spec in $(find "spec/$SPEC_SUITE" -type f -iname "*_spec.cr" | sort); do
   require="require \"./${spec##spec/}\""
 
-  if ! linker_command=$($CRYSTAL_BIN build --cross-compile --target x86_64--windows-msvc "$spec" 2>/dev/null); then
+  if ! linker_command=$($CRYSTAL_BIN build --cross-compile --target x86_64--windows-msvc -Dwithout_zlib -Dwithout_openssl "$spec" 2>/dev/null); then
```

Comment:
Not sure I understand. Why don't you just add these flags to the build step in #8676? They're currently required to run the spec suite.

Reply:
I don't think they are required, though? The build passes as is

Reply:
i.e. I think the flags become required only since *this* PR. And it would be nice to have that reflected in this PR.

Reply:
They're required since this PR because it enables more specs on win32 and some only compile with `-Dwithout_zlib -Dwithout_openssl` because these libraries would be linked by default.
The idea is to enable as much code as possible on win32 so we get the most coverage right now. This means we need to exclude those parts that depend on zlib and openssl, which is indicated by the respective compile time flags. We could technically set these flags automatically when compiling for win32, but that requires a compiler change and eventually it should work on win32 too. Another alternative would be to expand all guard conditions to `flag?(:without_openssl) || flag?(:win32)`, but even that adds additional noise. Since it is only a temporary limitation, it's best to accommodate for that in the windows build tool, which is necessary anyway while we need to cross-compile to win32.

Reply:
This is all correct. Sorry for the confusion. I'm just saying this PR should also include an according change to [.github/workflows/win.yml](https://github.com/oprypin/crystal/blob/42ddeb074076d71dff3e9798058f6f59dd5daf93/.github/workflows/win.yml) -- when that's submitted.

Reply:
Sure 👍